crates/nu-std/lib/iter.nu

```diff
@@ -37,6 +37,43 @@ export def "iter find" [ # -> any | null
     }
 }
 
+# Returns the index of the first element that matches the predicate or
+# -1 if none
+#
+# # Invariant
+# > The closure has to return a bool
+#
+# # Example
+# ```nu
+# use std ["assert equal" "iter find-index"]
+#
+# let res = (
+#     ["iter", "abc", "shell", "around", "nushell", "std"]
+#     | iter find-index {|x| $x starts-with 's'}
+# )
+# assert equal $res 2
+#
+# let is_even = {|x| $x mod 2 == 0}
+# let res = ([3 5 13 91] | iter find-index $is_even)
+# assert equal $res -1
+# ```
+export def "iter find-index" [ # -> int
+    fn: closure                # the closure used to perform the search
+] {
+    try {
```

Comment:
this could also be written without the `try` / `catch` as
```bash
    let matches = (
        enumerate
        | each {|it|
            if (do $fn $it.item) {
                $it.index
            }
        }
    )

    if ($matches | is-empty) {
        -1
    } else {
        $matches | first
    }
```

:wink:

Reply:
i do not have a strong preference here, pick the one you prefer :relieved: 
maybe i personally find the `if` easier to read? not sure

Reply:
yeah, this is more readable